fw/websocket.c

```diff
@@ -197,7 +197,6 @@ tfw_http_websocket_upgrade(TfwSrvConn *srv_conn, TfwCliConn *cli_conn)
 	 */
 	cli_conn->pair = (TfwConn *)ws_conn;
 	tfw_connection_get(cli_conn->pair);
-	BUG_ON(atomic_read(&cli_conn->refcnt) < 2);
```

Comment:
I'm afraid that with this change we might observe referencing the dead client connection as in #1647 we tried to send a response on a dead client connection. This assertion was introduced in https://github.com/tempesta-tech/tempesta/pull/1797 and we can not drop a reference count on the client connection because we have not forwarded response on the connection (the comment above references `tfw_http_conn_drop()` for this logic).

Which function calls sequence leads to `cli_conn->refcnt < 2` and what's its value (I hope not 0)?

Reply:
I've added little bit details to commit message and PR description.